lib/oli_web/components/delivery/students/students.ex

```diff
@@ -145,7 +145,19 @@ defmodule OliWeb.Components.Delivery.Students do
         Enum.sort_by(students, fn student -> student.email end, sort_order)
 
       :last_interaction ->
-        Enum.sort_by(students, fn student -> student.last_interaction end, sort_order)
+        comparator =
+          case sort_order do
+            :asc ->
+              fn x, y -> DateTime.compare(x.last_interaction, y.last_interaction) != :gt end
+
+            :desc ->
+              fn x, y -> DateTime.compare(x.last_interaction, y.last_interaction) != :lt end
+          end
+
+        Enum.sort(
+          students,
+          comparator
+        )
```

Comment:
There is something weird happening here... I run the test with the previous code `Enum.sort_by(students, fn student -> student.last_interaction end, sort_order)` and your new code and both are making the test pass. Could you please check that?

Reply:
the reason why this test is passing in both is because the previous code only compare with hour and ignore the year, if you put for example ~U[2019-04-05 13:28:42Z] on the set_interation for user_4, the previous code is going to fail.

Reply:
@tomasferok could you please alter the test to reflect your changes. Maybe add a test failing and a test passing so that you cover the two possible cases.